resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -387,7 +446,7 @@ private[spark] class Client(
         .withLabels(driverKubernetesSelectors)
         .endMetadata()
       .withNewSpec()
-        .withType("NodePort")
+        .withType(if (useIngress) "ClusterIP" else "NodePort")
```

Comment:
Making the service type configurable on its own seems like a good idea, considering https://github.com/apache-spark-on-k8s/spark/issues/129 also. That way, we could isolate the concerns. Do you have ideas of how we can manage the complexity from the proliferation of commandline options configuring services? I'm not so sure that there is one setting that fits a majority of the use cases.